### ISSUE TITLE
Check whether we use features deprecated in the new pandas-2.1.0 (out yesterday)

### ISSUE THREAD

**Issue by christophfink** [r5py/r5py#356]
Also geopandas now raises quite a lot of `DeprecationWarning` and `FutureWarning`, we might have to postpone our 0.1.1 release slightly to wait that all upstream libraries have worked through the now deprecated pandas features

**Comment by christophfink** [comment]
see https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#deprecations

**Comment by christophfink** [comment]
Or we freeze the pandas version in our dependencies

**Comment by christophfink** [comment]
```
r5py/r5/base_travel_time_matrix_computer.py:120: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  return data_set.applymap(lambda x: numpy.nan if x == MAX_INT32 else x)
 ```

**Comment by christophfink** [comment]
Upstream (geopandas, collecting here, then reporting there in one batch): 
```
.virtualenv/lib/python3.11/site-packages/geopandas/base.py:3517: in equals
    return self._data.equals(other._data)
```

**Comment by christophfink** [comment]
```
lib/python3.11/site-packages/geopandas/geoseries.py:645: in apply
    result = super().apply(func, convert_dtype=convert_dtype, args=args, **kwargs)
```

**Comment by christophfink** [comment]
`E       The list of emitted warnings is: [ FutureWarning('The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.')].`

**Comment by christophfink** [comment]
I froze the pandas version in #357 , so we can move this issue to the next milestone

**Comment by christophfink** [comment]
This will be resolved when #366 is merged
